packages/dataviews/src/components/dataviews-layout/index.tsx

```diff
@@ -37,10 +37,14 @@ export default function DataViewsLayout( { className }: DataViewsLayoutProps ) {
 		isItemClickable,
 		renderItemLink,
 		empty = __( 'No results' ),
+		picker,
 	} = useContext( DataViewsContext );
 
-	const ViewComponent = VIEW_LAYOUTS.find( ( v ) => v.type === view.type )
-		?.component as ComponentType< ViewBaseProps< any > >;
+	const ViewComponent = VIEW_LAYOUTS.filter(
+		( v ) => Boolean( v.isPicker ) === Boolean( picker )
```

Comment:
At the moment I've added an extra `picker` property to the context. The only use is the filtering here so that regular `DataViews` can't render the pickerGrid layout, and `DataViewsPicker` also can't render the normal `grid`, `list` or `table` layouts.